engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -31,28 +37,42 @@ trait FlinkSource[T] extends Source[T] {
 
 }
 
+/**
+  * Support for Context initialization. By default initContext maps raw source event to "input" variable in Context.
+  * @tparam T - type of event that is generated by this source. This is needed to handle e.g. syntax suggestions in UI
+  */
+trait SourceContextTransformation[T] { self: Source[T] =>
+
+  def initContext(processId: String, taskName: String): InitContextFunction[T] = new InitContextFunction[T](processId, taskName)
+
+}
+
 /**
   * Typical source with methods specific for Flink, user has only to define Source
   * @tparam T - type of event that is generated by this source. This is needed to handle e.g. syntax suggestions in UI
   */
-trait BasicFlinkSource[T] extends FlinkSource[T] with ExplicitUidInOperatorsSupport {
+trait BasicFlinkSource[T] extends FlinkSource[T] with SourceTestSupport[T] with ExplicitUidInOperatorsSupport with SourceContextTransformation[T] {
```

Comment:
Can we use composition instead of inheritance for `SourceContextTransformation`?

Reply:
Fixed. See `FlinkIntermediateRawSource`.